crates/sncast/src/starknet_commands/account/create.rs

```diff
@@ -128,38 +128,33 @@ pub async fn create(
         );
         message.push_str(&deploy_command);
     }
-
-    if add_profile.is_some() {
-        if let Some(url) = &create.rpc.url {
-            let config = CastConfig {
-                url: url.clone(),
-                account: account.into(),
-                accounts_file: accounts_file.into(),
-                keystore,
-                ..Default::default()
-            };
-            let config_path = resolve_config_file();
-            add_created_profile_to_configuration(
-                create.add_profile.as_deref(),
-                &config,
-                &config_path,
-            )?;
-        } else {
-            unreachable!("Conflicting arguments should be handled in clap");
-        }
+    let mut add_profile_message =
+        String::from("--add-profile flag was not set. No profile added to snfoundry.toml");
+
+    if create.add_profile.is_some() {
+        let url =
+            create.rpc.url.clone().expect(
+                "the argument '--network' should not be used with '--add-profile' argument",
+            );
+
+        let config = CastConfig {
+            url,
+            account: account.into(),
+            accounts_file: accounts_file.into(),
+            keystore,
+            ..Default::default()
+        };
+        let config_path = resolve_config_file();
+        let profile_name = add_profile.expect("add_profile should be Some");
+        add_profile_message =
+            format!("Profile {profile_name} successfully added to {config_path}",);
+        add_created_profile_to_configuration(create.add_profile.as_deref(), &config, &config_path)?;
     }
```

Comment:
How about something like this?

```suggestion
    let add_profile_message = if create.add_profile.is_some() {
        let url =
            create.rpc.url.clone().expect(
                "the argument '--network' should not be used with '--add-profile' argument",
            );

        let config = CastConfig {
            url,
            account: account.into(),
            accounts_file: accounts_file.into(),
            keystore,
            ..Default::default()
        };
        let config_path = resolve_config_file();
        let profile_name = add_profile.expect("add_profile should be Some");
        let add_profile_message =
            format!("Profile {profile_name} successfully added to {config_path}",);
        add_created_profile_to_configuration(create.add_profile.as_deref(), &config, &config_path)?;
        add_profile_message
    } else {
        String::from("--add-profile flag was not set. No profile added to snfoundry.toml")
    };  
```